### DISCUSSION TITLE
How to calculate A'*A

### DISCUSSION THREAD

**Discussion by ZhangHuYi** [CERN/TIGRE#514] [Q&A]
I'm a student studying CT reconstruction and recently wanted to solve the objective function using the split-bregam framework, but one of the steps was to calculate A'*A, and I wanted to ask how I could solve this problem。

**Comment by AnderBiguri** [comment]
Hi!
The answer is that you can't.
The matrix A is too large to be stored in memory for CBCT so Tigre never
computes it explicitly. Not only that, but the transpose it's just an
approximation. So, using Tigre, you can even only compute A*x or A'*b, but
not just A or A'.
So, you could compute A'*A*x,  but not just A'*A.

**Comment by AnderBiguri** [comment]
That said, I do believe split bregman has been used for CT in a way that
you don't need such matrix, but I can't remember the reference now. Check
the literature.
